### PR TITLE
4 useful patches, plus a new opacity approach

### PR DESCRIPTION
**Pull request by bakulf**
The first 4 patches are needed. The last 3 implement a new algorithm for the opacity. Note that it is back compatible, but it introduces 2 new settings:

switcher.settings.client_opacity_value_in_focus - opacity for the client currently in focus
switcher.settings.client_opacity_value_selected - opacity for the selected client.

personally I use these values:

switcher.settings.client_opacity_value = 0
switcher.settings.client_opacity_in_focus = 0.2
switcher.settings.client_opacity_selected = 0.8

Basically you have a full control of the opacity values for any kind of client: focus, selected, anything else.

### CODE REVIEW

**Comment by troglobit** [comment]
Looks great! You're more than welcome to change the defaults and document the old settings :smiley:
